distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/common/downloads/ClientDownloader.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.test.itests.common.downloads;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+import javax.annotation.Nullable;
+
+import org.apache.http.ConnectionClosedException;
+import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
+import org.apache.http.impl.nio.client.HttpAsyncClients;
+
+import ddf.test.itests.AbstractIntegrationTest;
+
+/**
+ * Download client abstraction for code re-use within integration tests.
+ */
+public abstract class ClientDownloader {
+    private static final String CSW_STUB_SOURCE_ID = "cswStubServer";
+
+    final CloseableHttpAsyncClient client = HttpAsyncClients.createDefault();
+
+    private Throwable caughtException = null;
+
+    Future future = null;
+
+    public static SynchronousClientDownloader createSynchronous() {
+        return new SynchronousClientDownloader();
+    }
+
+    public static AsynchronousClientDownloader createAsynchronous() {
+        return new AsynchronousClientDownloader();
+    }
+
+    protected abstract String processResults() throws ExecutionException;
+
+    public abstract void startDownload(String metacardId);
+
+    public abstract void stopDownload();
+
+    public CloseableHttpAsyncClient getClient() {
+        return client;
+    }
+
+    public boolean hasException() {
```

Comment:
Do we want to add an assertion on this method in the test code since we always expect a `ConnectionClosedException` wrapped in an `ExecutionException` for the cancelled client?